### PR TITLE
Enable list from memory

### PR DESCRIPTION
**Pull request by wojtek-t**
Ref #15945

### CODE REVIEW

File: pkg/client/cache/reflector.go [hunk 1]
```diff
@@ -227,7 +227,10 @@ func (r *Reflector) ListAndWatch(stopCh <-chan struct{}) error {
 	resyncCh, cleanup := r.resyncChan()
 	defer cleanup()
 
-	options := unversioned.ListOptions{}
+	// Explicitly set "0" as resource version, as we can handle some
+	// delays in the reflector framework (those will be caught by
```
Comment (gmarek) [inline pkg/client/cache/reflector.go]: I'm not sure I understand the comment correctly:
// Explicitly set "0" as resource version, as we can handle some
// delays in the reflector framework and all changes will be caught by
// the watch eventually.

Comment (wojtek-t) [inline pkg/client/cache/reflector.go]: Howe about this one:

Explicitly set "0" as resource version - it's fine for the List() to be served from cache and potentially be delayed relative to etcd contents. Reflector framework will catch up via Watch() anyway.

?

Comment (gmarek) [inline pkg/client/cache/reflector.go]: s/anyway/eventually/.

**Comment by gmarek** [comment]
@k8s-bot unit test this

File: pkg/client/cache/reflector.go (hunk 1 above)
Comment (wojtek-t) [inline pkg/client/cache/reflector.go]: done.

**Comment by wojtek-t** [comment]
@gmarek - PTAL

**Comment by gmarek** [comment]
LGTM if tests will pass.

**Pull request merged**